invenio_resources/resources/record.py

```diff
@@ -0,0 +1,240 @@
+        # rec_uuid = request_context.request_args.get("pid_value")
+        # return self.controller.patch(rec_uuid)
+        pass
+
+    def delete(self):
+        """Delete an item."""
+        # make_response(self.resource.delete(request_context), 204)
+        # rec_uuid = request_context.request_args.get("pid_value")
+        # return self.controller.delete(rec_uuid)
+        pass
+
+    #
+    # Secondary Interface
+    #
+    def load_item_from_request(self, content_type):
+        try:
+            return self.config.item_loaders[content_type]()
+        except KeyError:
+            raise UnsupportedMediaRESTError(content_type)
+
+    def make_list_response(self, record_list, total, aggregations, http_code):
+
+        link_from_record = partial(
+            self.config.links_factory, endpoint=self.bp_name
+        )
+
+        response_ctx = {
+            "links": link_from_record,
+            "prettyprint": resource_requestctx.request_args.get(
+                "prettyprint", False
+            ),
+        }
+
+        serialized_search_result = self.config.list_serializers[
+            resource_requestctx.accept_mimetype
+        ].serialize_search(record_list, total, aggregations, response_ctx)
+
+        # # Generate links for self/prev/next
+
+        # endpoint = get_view_name(self.name, LIST_VIEW)
+        # location = url_for(".{}".format(endpoint), _external=True)
+
+        # FIXME: pagination
+        # urlkwargs.update(size=pagination["size"], _external=True)
+
+        # links = {}
+
+        # def _link(name):
+        #     urlkwargs.update(pagination["links"][name])
+        #     links[name] = url_for(endpoint, **urlkwargs)
+
+        # _link("self")
+        # if pagination["from_idx"] >= 1:
+        #     _link("prev")
+        # if pagination["to_idx"] < min(total, self.max_result_window):
+        #     _link("next")
+
+        response = make_response(serialized_search_result, 200)
+        response.mimetype = resource_requestctx.accept_mimetype
+        # Add location headers
+        # TODO
+
+        return response
+
+    # FIXME: The interface is `item` but `record` looks better here
+    def make_item_response(self, record, http_code):
+
+        link_from_record = partial(
+            self.config.links_factory, endpoint=self.bp_name
+        )
+
+        response_ctx = {
+            "links": link_from_record,
+            "prettyprint": resource_requestctx.request_args.get(
+                "prettyprint", False
+            ),
+        }
+        serialized_item = self.config.item_serializers[
+            resource_requestctx.accept_mimetype
+        ].serialize(record, response_ctx)
+
+        response = make_response(serialized_item, 201)
+        response.mimetype = resource_requestctx.accept_mimetype
+
+        # Add location headers
+        response.headers.extend(
+            {"location": record_link_from_pid(record.pid, self.bp_name)}
+        )
+
+        return response
```

Comment:
`prettyprint` is part of the flask reponse. It specifies the tabbing.

The `links`, I somehow assumed it would be part of the response no matter the type of serialization. Do we think that sort of answer (no links) would be used/needed? 

Once this question is answered, we get into the separation of concerns. There we need to ask ourselves: Are links part of the *record* that goes in the response, or part of the *response* per se. IMHO, the second one, since is part of the REST paradigm.